(discuss) Way to implement async read/write

NikVolf opened the pull request:
@alexcrichton 
how do you think, will this do the expected job (might be ineffective though) ?

#2

NikVolf committed:
experimental implementation

parasyte commented:
@alexcrichton @NikVolf I came to this crate through a roundabout way; learned of [`parity-tokio-ipc`](https://github.com/NikVolf/parity-tokio-ipc) from the atom/xray issue tracker. Because I am interested in a crate that provides a common async API for domain sockets and named pipes.

There isn't a whole lot of pressure to get this PR merged, or even the crate published, which I understand. The current state of things is going to make it difficult to collaborate; `parity-tokio-ipc` depends on @NikVolf's fork. So any PRs that might have to land on this crate would have to bubble through that fork to reach `parity-tokio-ipc`, etc.

So I wanted to raise the issue here: Is this PR acceptable to merge? Or should the fork take over and get published from there? Should the crate source be pulled in as a sub-crate of `parity-tokio-ipc` since this one hasn't been published in the first place? Are there any other options that might be missing? Can I add some of that pressure which seems to be lacking? :)

I am pretty confident that I'll be using `parity-tokio-ipc`. It seems like the most reasonable choice for my needs. But the current state of this dependency makes me a little uneasy.

alexcrichton commented:
Ah sorry for the delay! If @NikVolf or @parasyte y'all would like to take over this crate that may be best, I don't mind transferring repos if that's the case or feel free to use the fork! I would be ok updating the readme and/or archiving this project here as well to point to a different repo if you'd like

c0gent committed:
Migrate/update to `tokio` crate.

* Migrate from `tokio_core` to `tokio`.
* Bump version to 0.2.

NikVolf committed:
Update README.md

NikVolf committed:
Update README.md

NikVolf committed:
update also dependency sample

NikVolf committed:
remove doc upload

NikVolf committed:
Merge pull request #1 from poanetwork/master

Migrate from `tokio_core` to `tokio`.

c0gent committed:
Add handle argument back to `NamedPipe` creation methods.

Needed by `PollEvented2` to ensure a single consistent thread for use
by multiple pipe connections.

NikVolf committed:
Merge pull request #2 from poanetwork/master

Add `handle` argument back to `NamedPipe` creation methods.

NikVolf committed:
update build script

NikVolf committed:
use crates.io version

NikVolf committed:
polish and docs

NikVolf committed:
Merge pull request #3 from NikVolf/use-crates-ver

Use crates.io version for mio-named-pipes

NikVolf committed:
remove whitespace

NikVolf committed:
also alter metadata

NikVolf committed:
Merge pull request #4 from NikVolf/polish

Polish crate and add docs

NikVolf committed:
update metadata
